diff -Pru a/dput-0.9.6.2/debian/rules b/dput-0.9.6.2/debian/rules
--- a/dput-0.9.6.2/debian/rules	2010-05-21 13:44:26.000000000 +0800
+++ b/dput-0.9.6.2/debian/rules	2011-04-19 16:24:11.319706711 +0800
@@ -1,7 +1,7 @@
 #!/usr/bin/make -f
 
 PACKAGE := dput
-TMPDIR := debian/tmp
+TMPDIR ?= debian/tmp
 
 build: build-stamp
 build-stamp:
@@ -58,8 +58,6 @@
 	install --mode=0644 debian/conffiles $(TMPDIR)/DEBIAN
 	install --mode=0755 debian/prerm $(TMPDIR)/DEBIAN
 	install --mode=0755 debian/postinst $(TMPDIR)/DEBIAN
-	dpkg-gencontrol -isp
-	dpkg --build $(TMPDIR) ..
 
 # Build architecture-dependent files here.
 binary-arch: build
