From 27436201464c6eca356b2482b3edfe5ab37aaefe Mon Sep 17 00:00:00 2001
From: microcai <microcai@fedoraproject.org>
Date: Thu, 24 Mar 2011 01:49:12 +0800
Subject: [PATCH] fix logout problem

---
 js/ui/endSessionDialog.js |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/js/ui/endSessionDialog.js b/js/ui/endSessionDialog.js
index c019e2a..3d773e6 100644
--- a/js/ui/endSessionDialog.js
+++ b/js/ui/endSessionDialog.js
@@ -447,7 +447,7 @@ EndSessionDialog.prototype = {
         this._stopTimer();
         DBus.session.emit_signal('/org/gnome/SessionManager/EndSessionDialog',
                                  'org.gnome.SessionManager.EndSessionDialog',
-                                 signal, '', []);
+                                 'Confirmed', '', []);
     },
 
     _onOpened: function() {
-- 
1.7.4.1

