diff --git a/build/Makefile b/build/Makefile
--- a/build/Makefile
+++ b/build/Makefile
@@ -35,9 +35,9 @@ libwxTranslationHelper.a:
 		$(CXX) -c $(LIBWXTRANSLATIONHELPER_FLAGS) && ar rcs libwxTranslationHelper.a *.o && cd ../../build/
 
 install:
-	install -m 755 -o root -g root ../bin/$(APP_NAME) /usr/bin/
-	mkdir /usr/bin/art/ && chmod 755 /usr/bin/art/
-	install -m 644 -o root -g root ../bin/art/*.ico /usr/bin/art/
+	install -m 755 -o root -g root ../bin/$(APP_NAME) ${DESTDIR}/usr/bin/
+	mkdir ${DESTDIR}/usr/bin/art/ && chmod 755 ${DESTDIR}/usr/bin/art/
+	install -m 644 -o root -g root ../bin/art/*.ico ${DESTDIR}/usr/bin/art/
 	
 uninstall:
 	rm -f /usr/bin/$(APP_NAME)

